Tavisca.Bootcamp.LanguageBasics.Exercise3/Program.cs

```diff
@@ -1,47 +1,155 @@
+		{
+			Test(
+				new[] { 3, 4 }, 
+				new[] { 2, 8 }, 
+				new[] { 5, 2 }, 
+				new[] { "P", "p", "C", "c", "F", "f", "T", "t" }, 
+				new[] { 1, 0, 1, 0, 0, 1, 1, 0 });
+			Test(
+				new[] { 3, 4, 1, 5 }, 
+				new[] { 2, 8, 5, 1 }, 
+				new[] { 5, 2, 4, 4 }, 
+				new[] { "tFc", "tF", "Ftc" }, 
+				new[] { 3, 2, 0 });
+			Test(
+				new[] { 18, 86, 76, 0, 34, 30, 95, 12, 21 }, 
+				new[] { 26, 56, 3, 45, 88, 0, 10, 27, 53 }, 
+				new[] { 93, 96, 13, 95, 98, 18, 59, 49, 86 }, 
+				new[] { "f", "Pt", "PT", "fT", "Cp", "C", "t", "", "cCp", "ttp", "PCFt", "P", "pCt", "cP", "Pc" }, 
+				new[] { 2, 6, 6, 2, 4, 4, 5, 0, 5, 5, 6, 6, 3, 5, 6 });
+            //Console.ReadKey(true);
+		}
+
+		private static void Test(int[] protein, int[] carbs, int[] fat, string[] dietPlans, int[] expected)
+		{
+			var result = SelectMeals(protein, carbs, fat, dietPlans).SequenceEqual(expected) ? "PASS" : "FAIL";
+			Console.WriteLine($"Proteins = [{string.Join(", ", protein)}]");
+			Console.WriteLine($"Carbs = [{string.Join(", ", carbs)}]");
+			Console.WriteLine($"Fats = [{string.Join(", ", fat)}]");
+			Console.WriteLine($"Diet plan = [{string.Join(", ", dietPlans)}]");
+			Console.WriteLine(result);
+		}
+
+		public static List<int> FindAllIndex(int[] elements, List<int> indexLists, int elem) {
+			/* Generate all index of an element */
+			List<int> indexList = new List<int>();
+
+			foreach (int i in indexLists) {
+				if (elements[i] == elem) 
+				indexList.Add(i);
+			}
+
+			return indexList;
+		}
+
+		public static int GetMax(int[] elements, List<int> indexList) {
+
+             /* Find maximun within the tied element */
+			int max = elements[indexList[0]];
+
+			for (int i = 1; i < indexList.Count; i++) {
+				if (elements[indexList[i]] > max)
+				 max = elements[indexList[i]];
+			}
+
+			return max;
+		}
+		public static int GetMin(int[] elements, List<int> indexList) {
+			
+            /* Find minimum within the tied element */
+			int min = elements[indexList[0]];
+
+			for (int i = 1; i < indexList.Count; i++) {
+				if (elements[indexList[i]] < min) 
+				min = elements[indexList[i]];
+			}
+
+			return min;
+		}
+
+		public static int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
+		{  
+		    /************************************************************
+			 Idea is to Don't search for min/max in complete array from index 0,
+			 Instead make an indexed list which limits your search element for min/max
+			*************************************************************/
+			
+			int[] calories = new int[protein.Length];
+			int[] item = new int[dietPlans.Length];
+
+			for (int i = 0; i < protein.Length; i++) {
+				calories[i] = (protein[i] + carbs[i]) * 5 + fat[i] * 9;
+			}
+			
+			for (int i = 0; i < dietPlans.Length; i++) {
+				string dietPlan = dietPlans[i];
+				List<int> indexList = new List<int>();
+                int max, min;
+
+				if (dietPlan.Length == 0) {
+					item[i] = 0;
+				}
+

```

Comment:
there are way too many for loops few of them could have been avoided. Please refactor code to have only required for loops and use linq functions where necessary